src/include/storage/data_table.h

```diff
@@ -254,6 +377,7 @@ class DataTable {
   mutable common::SpinLatch blocks_latch_;
   // latch used to protect insertion_head_
   mutable common::SpinLatch header_latch_;
+  std::vector<tbb::concurrent_unordered_set<RawBlock *>> regions_;
```

Comment:
I think both std::list blocks_ and regions_ are stored the same information for now(although for a different purpose). Are there other ways to make memory usage more efficient?

Reply:
Added a comment about purpose, we didn't really have ideas to make memory usage more effect because the ordered blocks are required anyways for sequential scans.

Reply:
I don't know if there are any tests that rely on the order of which the blocks are iterated over. If not, then we can get rid of `blocks_` and instead use just `regions_`. That would be very nice.